### PR TITLE
JAMES-1652 Add JWT PublicKey in JMAP configuration

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Comment by aduprat** [comment]
test this please

File: server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java [hunk 1]
```diff
@@ -42,14 +45,23 @@ protected void configure() {
...
 
     private PropertiesConfiguration getConfiguration(FileSystem fileSystem) throws FileNotFoundException, ConfigurationException {
         return new PropertiesConfiguration(fileSystem.getFile(FileSystem.FILE_PROTOCOL_AND_CONF + "jmap.properties"));
     }
+
+    private String loadPublicKey(FileSystem fileSystem, String jwtPublickeyUrl) throws FileNotFoundException, IOException {
+        if (!Strings.isNullOrEmpty(jwtPublickeyUrl)) {
+            return FileUtils.readFileToString(fileSystem.getFile(jwtPublickeyUrl));
+        }
+        return null;
```
Comment (mbaechler) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: why not optional ?

Comment (aduprat) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: OK

**Comment by rouazana** [comment]
:+1:

File: server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java [hunk 2]
```diff
@@ -42,14 +45,23 @@ protected void configure() {
...
+                .secret(configuration.getString("tls.secret"))
+                .jwtPublicKeyPem(loadPublicKey(fileSystem, Optional.ofNullable(configuration.getString("jwt.publickeypem.url"))))
+                .build();
     }
 
     private PropertiesConfiguration getConfiguration(FileSystem fileSystem) throws FileNotFoundException, ConfigurationException {
         return new PropertiesConfiguration(fileSystem.getFile(FileSystem.FILE_PROTOCOL_AND_CONF + "jmap.properties"));
     }
+
+    private String loadPublicKey(FileSystem fileSystem, Optional<String> jwtPublickeyPemUrl) throws FileNotFoundException, IOException {
```
Comment (mbaechler) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: why not Optional here ?

Comment (aduprat) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: No, we don't mind when reading the PropertiesConfiguration elements if an element is given or not; this is delegated to the JMAPConfiguration builder.

Comment (mbaechler) [inline server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java]: Sorry but it doesn't make sense : you have an optional input url but a null string as an output ...

File: server/container/jetty/pom.xml [hunk 3]
```diff
@@ -129,6 +129,11 @@
             </activation>
             <dependencies>
                 <dependency>
+                    <groupId>com.github.fge</groupId>
+                    <artifactId>throwing-lambdas</artifactId>
+                    <version>0.5.0</version>
```
Comment (mbaechler) [inline server/container/jetty/pom.xml]: please put the version in parent

Comment (aduprat) [inline server/container/jetty/pom.xml]: Done

File: server/pom.xml [hunk 4]
```diff
@@ -193,6 +193,7 @@
         <cassandra-unit.version>2.1.9.2</cassandra-unit.version>
         <assertj-1.version>1.7.1</assertj-1.version>
         <assertj-3.version>3.2.0</assertj-3.version>
+        <throwing-lambdas.version>0.5.0</throwing-lambdas.version>
```
Comment (mbaechler) [inline server/pom.xml]: usually we use dependencyManagement section for declaring versions, not properties

Comment (aduprat) [inline server/pom.xml]: OK

**Comment by mbaechler** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Pull request closed without merging**